packages/loader/src/deterministic.ts

```diff
@@ -0,0 +1,52 @@
+import type { GenesisCommit, GenesisHeader } from '@ceramicnetwork/common'
+import { ModelInstanceDocument } from '@ceramicnetwork/stream-model-instance'
+import { StreamID } from '@ceramicnetwork/streamid'
+import * as codec from '@ipld/dag-cbor'
+import { CID } from 'multiformats/cid'
+import { sha256 } from 'multiformats/hashes/sha2'
+
+import type { LoadKey } from './types.js'
+
+export type GenesisMetadata = {
+  controller: string
+  model: StreamID | string
+  unique?: Array<string>
+}
+
+/**
+ * Create a deterministic genesis commit matching the logic unsed internally by ModelInstanceDocument.
+ */
+export function createDeterministicGenesis(meta: GenesisMetadata): GenesisCommit {
+  const model = meta.model instanceof StreamID ? meta.model : StreamID.fromString(meta.model)
+  const header: GenesisHeader = { controllers: [meta.controller], model: model.bytes, sep: 'model' }
+  if (Array.isArray(meta.unique)) {
+    header.unique = new TextEncoder().encode(meta.unique.join('|'))
```

Comment:
Yes, it provides the same output, that's tested in https://github.com/ceramicstudio/js-composedb/pull/207/files#diff-9416770bd4e5c5106a2eed5c312a1ed7f4edeb74ea3736db6b76e8b8daa3b050

I agree though it would be nice to have this directly available from the MID package, essentially there are 2 things needed here:

1. Create the genesis commit based on the input (this function)
2. Get the stream ID for the commit: https://github.com/ceramicstudio/js-composedb/pull/207/files#diff-0af4f6bd66935ecbd445ba22995dd687aeeee5468afdd46d060f8543a4a9a195R31

Do you see both functions as being part of the MID package or only the first one please? Shall we expose the stream ID from genesis function somewhere else?

Reply:
The former could certainly be made a public method in ModelInstanceDocument.

The latter, to get the StreamID from the genesis commit contents, that's not really specific to ModelInstanceDocuments.  I'd be okay with exposing a utility function from a js-ceramic package to do this, just need to think of what packages makes the most sense for it.  Maybe the StreamID package actually, though that would require pulling in the dag-jose codec into the streamid package which might bloat the package size...